pkg/messaging/v1/util.go

```diff
@@ -277,3 +314,43 @@ func ErrorFromInternalStatus(internalStatus *internalv1pb.Status) error {
 
 	return grpc_status.ErrorProto(respStatus)
 }
+
+func processGRPCToHTTPTraceHeaders(ctx context.Context, traceContext string, setHeader func(string, string)) {
+	// attach grpc-trace-bin value in traceparent and tracestate header
+	decoded, _ := base64.RawStdEncoding.DecodeString(traceContext)
+	sc, ok := propagation.FromBinary(decoded)
+	if !ok {
+		sc = diag.FromContext(ctx)
+	}
+	diag.SpanContextToHTTPHeaders(sc, setHeader)
+}
+
+func processHTTPToHTTPTraceHeaders(ctx context.Context, traceparentValue, traceStateValue string, setHeader func(string, string)) {
+	if traceparentValue == "" {
+		sc := diag.FromContext(ctx)
+		diag.SpanContextToHTTPHeaders(sc, setHeader)
+	} else {
+		setHeader(traceparentHeader, traceparentValue)
+		setHeader(tracestateHeader, traceStateValue)
+	}
+}
+
+func processHTTPToGRPCTraceHeader(ctx context.Context, md metadata.MD, traceparentValue, traceStateValue string) {
+	var sc trace.SpanContext
+	var ok bool
+	if sc, ok = diag.SpanContextFromString(traceparentValue); ok {
+		sc.Tracestate = diag.TraceStateFromString(traceStateValue)
+	} else {
+		sc = diag.FromContext(ctx)
+	}
+	md.Append(tracebinMetadata, base64.RawStdEncoding.EncodeToString(propagation.Binary(sc)))
+}
+
+func processGRPCToGRPCTraceHeader(ctx context.Context, md metadata.MD, grpctracebinValue string) {
+	if grpctracebinValue == "" {
+		sc := diag.FromContext(ctx)
+		md.Append(tracebinMetadata, base64.RawStdEncoding.EncodeToString(propagation.Binary(sc)))
```

Comment:
I would use`md.Set()` instead of using md.Append(). 

Seems like grpc doesn't decode grpc-tracebin automatically. is this correct?

In this case, caller will get base64 encoded `grpc-trace-bin`. so that caller needs to decode base64 encoded grpc-trace-bin.

Reply:
done. Yeah, I am testing it around that to conclude, it should just work ..

Reply:
I am hitting the following issue if  I don't use encoding.  It should work without explicit encoding.

panic: rpc error: code = Internal desc = grpc: error while marshaling: proto: field "dapr.proto.internals.v1.ListStringValue.Values" contains invalid UTF-8